test/metabase/driver/sql_jdbc/connection_test.clj

```diff
@@ -75,50 +75,64 @@
         ;; ensure that, for any sql-jdbc drier anyway, we found *some* DB name to use in this String
         (is (not= db-nm "null"))))))
 
+(deftest same-connection-details-result-in-equal-specs-test
+  (testing "Two JDBC specs created with the same details must be considered equal for the connection pool cache to work correctly"
+    ;; this is only really a concern for drivers like Spark SQL that create custom DataSources instead of plain details
+    ;; maps -- those DataSources need to be considered equal based on the connection string/properties
+    (mt/test-drivers (sql-jdbc.tu/sql-jdbc-drivers)
+      (let [details (:details (mt/db))
+            spec-1  (sql-jdbc.conn/connection-details->spec mt/*driver* details)
+            spec-2  (sql-jdbc.conn/connection-details->spec mt/*driver* details)]
+        (is (= spec-1 spec-2))))))
```

Comment:
Added a test to make sure that two specs created with identical details are considered equal. This test will run for all SQL JDBC drivers. This only really affects drivers that return a `{:datasource javax.sql.DataSource}` spec since just reifying it will not result in the objects being considered equal even if the underlying connection URL and properties are. A record type is more appropriate here; this test should catch situations where we forget to use them